sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -31,7 +32,7 @@ use sway_error::{
 };
 use sway_types::{integer_bits::IntegerBits, Ident, Span, Spanned};
 
-use std::collections::{HashMap, VecDeque};
+use std::collections::{BTreeSet, HashMap, VecDeque};
```

Comment:
From what I could understand, the order itself doesn't matter, so maybe [`FxHashSet`](https://github.com/FuelLabs/sway/issues/3162) could be faster?

Reply:
`FxHashMap`s also are deterministic. (though in undefined order). The issue I linked to discusses this.

Reply:
Awesome! That'll be very handy.